### PR TITLE
Issue 22297 macro

### PR DESCRIPTION
**Pull request by jdotcms**
Refreshing macro cache when flushing it

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/MacroCacheRefresherJob.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+     * Refresh the macros contained in the fileAsset
+     * @param fileAsset {@link FileAsset}
+     */
+    public void refreshFileAsset (final FileAsset fileAsset) {
+
+        try (InputStream in = fileAsset.getInputStream()) {
+
+            final String velocityCode = IOUtils.toString(in, Charset.defaultCharset());
+            VelocityUtil.getInstance().parseVelocity(velocityCode, com.dotmarketing.util.VelocityUtil.getBasicContext());
+        } catch (IOException e) {
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/MacroCacheRefresherJob.java:107]: Let's include the file's name and/or identifier in the error message

File: dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileListener.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+     */
+    default String getId () {
+        return this.getClass().getName();
+    }
+
+    /**
+     * When a file that is being modified this event is triggered, match the listener criteria,
+     * @param fileEvent {@link FileEvent}
+     */
+    default void fileModify(FileEvent fileEvent) {}
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileListener.java:21]: Should this method be like `onFileChanged` or something like that?

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileListener.java (hunk 2 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileListener.java:21]: yeah but the FolderListener is in the same way, so I leave as it is to keep the consistence at least

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java [hunk 3]
```diff
@@ -153,6 +153,42 @@ public final Response currentSite(@Context final HttpServletRequest httpServletR
...
     }
 
+    /**
+     * Returns the default site
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @return
+     */
+    @GET
+    @Path ("/defaultSite")
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java:163]: excelent!

**Review by dsilvam — APPROVED** [review]

**Comment by jdotcms** [comment]
Postman are not finishing merging with master and running again

**Pull request merged**